webapp/components/_new/generic/BaseButton/BaseButton.vue

```diff
@@ -0,0 +1,144 @@
+<template>
+  <button
+    :class="buttonClass"
+    :disabled="loading"
+    :type="type"
+    @click.capture="event => $emit('click', event)"
+  >
+    <base-icon v-if="icon" :name="icon" />
+    <loading-spinner v-if="loading" />
+    <slot />
+  </button>
+</template>
+
+<script>
+import LoadingSpinner from '~/components/_new/generic/LoadingSpinner/LoadingSpinner'
+
+export default {
+  components: {
+    LoadingSpinner,
+  },
+  props: {
+    circle: {
+      type: Boolean,
+      default: false,
+    },
+    danger: {
+      type: Boolean,
+      default: false,
+    },
+    filled: {
+      type: Boolean,
+      default: false,
+    },
+    ghost: {
+      type: Boolean,
+      default: false,
+    },
+    icon: {
+      type: String,
+    },
+    loading: {
+      type: Boolean,
+      default: false,
+    },
+    size: {
+      type: String,
+      default: 'regular',
+      validator(value) {
+        return value.match(/(small|regular)/)
+      },
+    },
+    type: {
+      type: String,
+      default: 'button',
+      validator(value) {
+        return value.match(/(button|submit)/)
+      },
+    },
+  },
+  computed: {
+    buttonClass() {
+      let buttonClass = 'base-button'
+
+      if (this.$slots.default === undefined) buttonClass += ' --icon-only'
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-01-14T17:55:25Z" title="Tuesday, January 14th 2020, 6:55:25 pm +01:00">Jan 14, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,144 @@
+<template>
+  <button
+    :class="buttonClass"
+    :disabled="loading"
+    :type="type"
+    @click.capture="event => $emit('click', event)"
+  >
+    <base-icon v-if="icon" :name="icon" />
+    <loading-spinner v-if="loading" />
+    <slot />
+  </button>
+</template>
+
+<script>
+import LoadingSpinner from '~/components/_new/generic/LoadingSpinner/LoadingSpinner'
+
+export default {
+  components: {
+    LoadingSpinner,
+  },
+  props: {
+    circle: {
+      type: Boolean,
+      default: false,
+    },
+    danger: {
+      type: Boolean,
+      default: false,
+    },
+    filled: {
+      type: Boolean,
+      default: false,
+    },
+    ghost: {
+      type: Boolean,
+      default: false,
+    },
+    icon: {
+      type: String,
+    },
+    loading: {
+      type: Boolean,
+      default: false,
+    },
+    size: {
+      type: String,
+      default: 'regular',
+      validator(value) {
+        return value.match(/(small|regular)/)
+      },
+    },
+    type: {
+      type: String,
+      default: 'button',
+      validator(value) {
+        return value.match(/(button|submit)/)
+      },
+    },
+  },
+  computed: {
+    buttonClass() {
+      let buttonClass = 'base-button'
+
+      if (this.$slots.default == null) buttonClass += ' --icon-only'
```


why do you use `==` instead of `===` here?

Reply:
> [<img alt="alina-beck" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/alina-beck) **Authored by [alina-beck](https://github.com/alina-beck)**
_<time datetime="2020-01-15T08:01:15Z" title="Wednesday, January 15th 2020, 9:01:15 am +01:00">Jan 15, 2020</time>_

---


To also return `true` when it's `undefined`. But I just double-checked and I think slots are `undefined` when they're empty so it could be `=== undefined` instead. :+1: